lib/model/store.dart

```diff
@@ -230,6 +230,11 @@ class PerAccountStore extends ChangeNotifier {
           userSettings?.twentyFourHourTime        = event.value as bool;
         case UserSettingName.displayEmojiReactionUsers:
           userSettings?.displayEmojiReactionUsers = event.value as bool;
+        case UserSettingName.emojiset:
+          final value = event.value as Emojiset?;
+          if (value != null) {
+            userSettings?.emojiset                = value;
```

Comment:
What's the case where this is null? I think it's that the value was unknown.

In that case it doesn't seem right to leave this at whatever the previous value happened to be — the hysteresis would be a bit odd.

Probably cleanest is to have an "unknown" value in the enum, and have `Emojiset.fromRawString` return that in that case. Then when we go to use the value to show an emoji, we can treat "unknown" the same as some default like "google".

Alternatively, we could just throw an exception so the event fails to deserialize. That's what this PR already does in the initial snapshot, I think, which is going to be the much more frequent place to encounter that if we ever do.

Reply:
Makes sense. In my next revision, I plan to have it throw an exception. I'm not sure there's a nice way to designate a new `unknown` enum value for internal use only. We don't want to complicate the question of what happens when the server actually gives us the string "unknown".

Reply:
Cool, SGTM.

Just for the sake of when discussing other enums, though, where we may have stronger motivation to handle unknown values:

> We don't want to complicate the question of what happens when the server actually gives us the string "unknown".

I think this actually would work out smoothly on its own: if the server gives us "unknown", we'd look that up in the map and map it to the enum value `Emojiset.unknown`. Which is exactly the right answer, even though it got there in a different way — "unknown" is indeed a value we didn't know about for this enum.

Now if at some point the server/API added "unknown" as a valid value, and then we added support for that, we'd have to do something to handle it. Probably simplest is we'd just pick some other name to represent this sense of "unknown". That'd mean that `Emojiset.unknown` would mean one thing (namely, an enum value that we didn't know about when writing the code) in an earlier version of our code, and a different thing (namely, an enum value that the server represents with the string "unknown", presumably meaning that the server doesn't know the right value or something like that) in a later version of our code. But that's fine — it's just a refactor within our code. Thankfully all of the APIs that we define at the Dart level are completely within our control, with nothing external consuming them, and we're free to refactor them at will.

(Even if/when we start storing the user's server data to disk, at worst this sort of change would mean writing a migration. And probably in fact we'd continue the practice we had in zulip-mobile with `dropCache`, so that for server data we'd just drop it and re-fetch rather than migrate it.)

Reply:
Interesting; yeah, I guess that makes sense.